app/src/main/java/com/duckduckgo/app/browser/omnibar/OmnibarFeatureFlagObserver.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.omnibar
+
+import androidx.lifecycle.LifecycleOwner
+import com.duckduckgo.app.browser.omnibar.model.OmnibarPosition
+import com.duckduckgo.app.di.AppCoroutineScope
+import com.duckduckgo.app.lifecycle.MainProcessLifecycleObserver
+import com.duckduckgo.app.settings.db.SettingsDataStore
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesMultibinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@ContributesMultibinding(
+    scope = AppScope::class,
+    boundType = MainProcessLifecycleObserver::class,
+)
+@SingleInstanceIn(AppScope::class)
+class OmnibarFeatureFlagObserver @Inject constructor(
```

Comment:
This is not the right place to do this. On process start will get called before a privacy config has been downloaded. If the feature flag value in remote has changed, you won’t get the `settingsDataStore` updated until the next time the app cold starts.

What you want to use is `PrivacyConfigCallbackPlugin`. This plugin will notify you when the privacy config has been downloaded, and will let you update the position if necessary.

Check https://github.com/duckduckgo/Android/blob/develop/experiments/experiments-impl/src/main/java/com/duckduckgo/experiments/impl/loadingbarexperiment/LoadingBarExperimentVariantInitializer.kt as an example.

Reply:
Ah, good to know, thanks. I’ve fixed it [here](https://github.com/duckduckgo/Android/pull/4885/commits/a9fa84366fa19b58df6f8029c98e084d29f3941f).